src/tests/ftest/recovery/check_start_options.py

```diff
@@ -156,3 +143,93 @@ def test_check_start_reset(self):
         dmg_command.check_disable()
         # The pool is orphan pool, so skip the cleanup.
         pool.skip_cleanup()
+
+    def test_check_start_failout(self):
+        """Test dmg check start --failout=on.
+
+        See the state diagram attached to the ticket.
+
+        1. Create a pool.
+        2. Inject orphan pool fault.
+        3. Enable checker and set policy to --all-interactive.
+        4. Start the checker with --failout=on.
+        5. Query and check that it detected the orphan pool.
+        6. Remove the pool directory from the mount point.
+        7. Repair the pool.
+        8. Query and verify that Current status is FAILED. i.e., State is failed.
+
+        Jira ID: DAOS-17818
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=vm
+        :avocado: tags=recovery,cat_recov
+        :avocado: tags=DMGCheckStartOptionsTest,test_check_start_failout
+        """
+        # 1. Create a pool.
+        self.log_step("Create a pool")
+        pool = self.get_pool(connect=False)
+
+        # 2. Inject orphan pool fault.
+        self.log_step("Inject orphan pool fault")
+        dmg_command = self.get_dmg_command()
+        dmg_command.faults_mgmt_svc_pool(
+            pool=pool.identifier, checker_report_class="CIC_POOL_NONEXIST_ON_MS")
+
+        # 3. Enable checker and set policy to --all-interactive.
+        self.log_step("Enable checker and set policy to --all-interactive.")
+        dmg_command.check_enable()
+        dmg_command.check_set_policy(all_interactive=True)
+
+        # 4. Start the checker with --failout=on.
+        self.log_step("Start the checker with --failout=on.")
+        dmg_command.check_start(failout="on")
+
+        # 5. Query and check that it detected the orphan pool.
+        self.log_step("Query and check that it detected the orphan pool.")
+        query_reports = self.query_detect(fault="orphan pool")
+
+        # 6. Remove the pool directory from the mount point.
+        self.log_step("Remove the pool directory from the mount point.")
+        scm_mount = self.server_managers[0].get_config_value("scm_mount")
+        pool_path = f"{scm_mount}/{pool.uuid.lower()}"
+        pool_out = check_file_exists(
+            hosts=self.hostlist_servers, filename=pool_path, sudo=True)
+        if not pool_out[0]:
+            msg = ("MD-on-SSD cluster. Contents under mount point are removed by control "
+                   "plane after system stop.")
+            self.log.info(msg)
+            dmg_command.system_start()
+            # return results in PASS.
+            return
+        command = f"sudo rm -rf {pool_path}"
+        remove_result = run_remote(
+            log=self.log, hosts=self.hostlist_servers, command=command)
+        if not remove_result.passed:
+            self.fail(f"Failed to remove {pool_path} from {self.hostlist_servers}")
```

Comment:
You could print just the hosts it failed on

```suggestion
        if not remove_result.passed:
            self.fail(f"Failed to remove {pool_path} from {remove_result.failed_hosts}")
```

Reply:
Done